CHANGELOG.md

```diff
@@ -10,11 +10,16 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
-## Unreleased
+## [1.21.0](https://github.com/uber-go/fx/compare/v1.20.1...v1.21.0) - 2024-03-13
 
 ### Added
 - fxtest: Add WithTestLogger option that uses a `testing.TB` as the
   Fx event logger.
+- An fxevent logger that can log events using a slog logger has been added.
+
+### Changed
+- Upgrade Dig dependency to v1.17.1
+- Update and clarify a lot of the API documentation
```

Comment:
Can we remove this? We should only contain user facing functional changes in the changelog.